specification/refarch.adoc

```diff
@@ -3,142 +3,184 @@
-looked up to derive a confidential | non-confidential physical memory attribute. 
-This lookup should be performed for the page sizes per the paging modes supported by the hart.
+The figure 2 below describes a reference model for memory tracking lookup where
+the physical address derived from the first and guest stage nested page walk is
+looked up via the MTT configured for the active supervisor domain to get the
+access permissions for the physical address. This lookup should be performed
+for the page sizes per the paging modes supported by the hart.
 
 [caption="Figure {counter:image}: ", reftext="Figure {image}"]
-[title= "Memory Tracking for Confidential PMA"]
+[title= "Memory Tracking for Supervisor Domains"]
 image::img_8.png[]
-	
-==== Management of Confidential | Non-Confidential dynamic Physical Memory Attributes
-
-The SW TCB manages the assignment of the confidential PMA to memory regions, 
-while the HW TCB enforces the confidential PMA for TVM accesses. The region sizes 
-at which the memory tracking enforces confidential properties may be multiples of 
-the architectural page sizes supported by the hart MMU. Similarly, the IOMMU should 
-support a similar memory tracking lookup to enable direct device access into TVM memory 
-regions. For the CoVE reference architecture this TCB hence consists of the HW 
-(MMU, IOMMU, Memory Controller) and the SW/FW elements - TSM-driver and the TSM. 
-The TSM is responsible for enforcing isolation of confidential memory pages among TVMs 
-(via G-stage translation) - pages assigned to the TVM may belong to C | NC PMA to 
-allow for IO access. The TSM may manage additional attributes on TVM-assigned pages such as: 
-TVM-owner, Page-sub-type, TLB versioning information, Locking semaphore and additional metadata etc.
-This extended memory tracking information managed by the TSM is referred to as the EMTT.
-
-==== Handling Implicit, Explicit Accesses
-For TVM accesses for instruction fetch and page walks (virtual address 
-translation), a Confidential PMA is required to enforce the following security properties:
-
-* TEE Instruction fetch - security property: TVM/TSM cannot fetch code 
-from untrusted shared memory
-* TEE Paging structure walk - security property: TVM/TSM cannot locate 
-page tables in untrusted shared memory (for TVM, G-stage walks enforced 
+
+==== Management of isolation for Confidential Physical Memory
+
+The SW TCB (TSM) manages the assignment of physical memory to the Confidential
+supervisor domain, while the HW TCB (hart MMU including Sv, MTT Extensions)
+enforces the access-control for confidential memory against other supervisor
+domains. The region sizes at which the memory tracking enforces isolation may
+be multiples of the architectural page sizes supported by the hart MMU.
+Similarly, the IOMMU via the IOMTT is expected to support a similar memory
+tracking lookup to enable direct IOMMU and IOMMU-managed device access into
```

Comment:
```suggestion
Similarly, the IOMMU is expected to support a similar memory tracking lookup 
to enable a device/function trusted by the TVM to directly access 
```

Reply:
The IOMMU is not involved with IOMTT or similar memory tracking. 
A better text is 
"Similarly, MTT enforcement for memory accesses originated by I/O devices and IOMMUs"